|Field | description | type | properties | example | enum|
| ---| ---| ---| ---| ---| --- |
| [alternateBases](./obj/alternateBases.md) | Alternate bases for this variant (starting from `start`). * Accepted values: IUPAC codes for nucleotides (e.g. `https://www.bioinformatics.org/sms/iupac.html`). N is a wildcard, that denotes the position of any base, and can beused as a standalone base of any type or within a partially knownsequence. As example, a query of `ANNT` the Ns can take take any form of[ACGT] and will match `ANNT`, `ACNT`, `ACCT`, `ACGT` ... and so forth.* an *empty value* is used in the case of deletions with the maximally trimmed, deleted sequence being indicated in `ReferenceBases`* Categorical variant queries, e.g. such *not* being represented through sequence & position, make use of the `variantType` parameter. * Either `alternateBases` or `variantType` is required. | string | NA | NA | NA|
| [caseLevelData](./obj/caseLevelData.md) | caseLevelData reports about the variation instances observed in individual analyses. | array | [alleleOrigin](./obj/alleleOrigin.md), [analysisId](./obj/analysisId.md), [biosampleId](./obj/biosampleId.md), [clinicalInterpretations](./obj/clinicalInterpretations.md), [id](./obj/id.md), [individualId](./obj/individualId.md), [phenotypicEffects](./obj/phenotypicEffects.md), [runId](./obj/runId.md), [zygosity](./obj/zygosity.md) | NA | NA|
| [frequencyInPopulations](./obj/frequencyInPopulations.md) | NA | array | [frequencies](./obj/frequencies.md), [source](./obj/source.md), [sourceReference](./obj/sourceReference.md), [version](./obj/version.md) | NA | NA|
| [identifiers](./obj/identifiers.md) | NA | object | [clinVarIds](./obj/clinVarIds.md), [genomicHGVSId](./obj/genomicHGVSId.md), [proteinHGVSIds](./obj/proteinHGVSIds.md), [transcriptHGVSIds](./obj/transcriptHGVSIds.md), [variantAlternativeIds](./obj/variantAlternativeIds.md) | NA | NA|
| [molecularAttributes](./obj/molecularAttributes.md) | NA | object | [aminoacidChanges](./obj/aminoacidChanges.md), [geneIds](./obj/geneIds.md), [genomicFeatures](./obj/genomicFeatures.md), [molecularEffects](./obj/molecularEffects.md) | NA | NA|
| [position](./obj/position.md) | This section groups all attributes that allows to identify a variant via its position in the genome. | object | [assemblyId](./obj/assemblyId.md), [end](./obj/end.md), [refseqId](./obj/refseqId.md), [start](./obj/start.md) | NA | NA|
| [referenceBases](./obj/referenceBases.md) | Reference bases for this variant (starting from `start`). * Accepted values: IUPAC codes for nucleotides (e.g. `https://www.bioinformatics.org/sms/iupac.html`). N is a wildcard, that denotes the position of any base, and can be used as a standalone base of any type or within a partially known sequence. As example, a query of `ANNT` the Ns can take take any form of `[ACGT]` and will match `ANNT`, `ACNT`, `ACCT`, `ACGT` ... and so forth. * an *empty value* is used in the case of insertions with the maximally trimmed, inserted sequence being indicated in `AlternateBases`. NOTE: Many Beacon instances could not support UIPAC codes and it is not mandatory for them to do so. In such cases the use of [ACGTN] is mandated. | string | NA | NA | NA|
| [variantInternalId](./obj/variantInternalId.md) | Reference to the **internal** variant ID. This represents the primary key/identifier of that variant **inside** a given Beacon instance. Different Beacon instances may use identical id values, referring to unrelated variants. Public identifiers such as the GA4GH Variant Representation Id (VRSid) MUST be returned in the `identifiers` section. A Beacon instance can, of course, use the VRSid as their own internal id but still MUST represent this then in the `identifiers` section. | string | NA | NA | NA|
| [variantLevelData](./obj/variantLevelData.md) | NA | object | [clinicalInterpretations](./obj/clinicalInterpretations.md), [phenotypicEffects](./obj/phenotypicEffects.md) | NA | NA|
| [variantType](./obj/variantType.md) | The `variantType` declares the nature of the variation in relation to a reference. In a response, it is used to describe the variation. In a request, it is used to declare the type of event the Beacon client is looking for. If in queries variants can not be defined through a sequence of one or more bases (`precise` variants) it can be used standalone (i.e. without `alternateBases`) together with positional parameters. Examples here are e.g. queries for structural variants such as `DUP` (increased allelic count of material from the genomic region  between `start` and `end` positions without assumption about the placement of the additional sequence) or `DEL` (deletion of sequence following `start`). Either `alternateBases` or `variantType` is required, with the exception of range queries (single `start` and `end` parameters). | string | NA | SNP, DEL, DUP, BND | NA|
